src/http/server/response.cr

```diff
@@ -40,6 +40,30 @@ class HTTP::Server
       output.response = self
     end
 
+    def remote_address
+      case @io
+      when IPSocket
+        io = @io
+      when UNIXSocket
+        io = @io
+      when OpenSSL::SSL::Socket
+        io = @io.as(OpenSSL::SSL::Socket).io
```

Comment:
No need for calling `.as`, ditto below.

Reply:
Unfortunately there appears to be an issue with type inference, removing the cast results in:
```
in src/http/server/response.cr:50: undefined method 'io' for IO::ARGF (compile-time type is IO+)

        io = @io.io
```